Keepers [Don't Merge]

Petethegoat opened the pull request:
Keepers second draft. Still plenty left to do. Wildly untested and unfinished. Lots of quick code.
Features right now are:
- Two hands that work as human hands do (or monkey hands, to be more accurate. They can't use computers.)
- Binary speech only.
- Vent crawling.
- All access.
- Basic hud.
- Laws, yo
- Keeper cannibalism.

Still need to add spawning and emagging. Maybe more stuff with speech. Maybe tools and shit.

Minor refactoring of proc/swap_hand(), definition is moved to /mob.
Adjustments to hotkeys.
Added var/access_card to simple_animal, they can now utilize access.

Petethegoat committed:
Keepers first draft. Still plenty left to do.
Minor refactoring of proc/swap_hand(), definition is moved to /mob.
Adjustments to hotkeys.
Added var/access_card to simple_animal, they can now utilize access.

Petethegoat commented on code/modules/mob/living/simple_animal/friendly/keeper.dm:
```diff
@@ -0,0 +1,111 @@
+	return
+
+
+/mob/living/simple_animal/keeper/put_in_l_hand(obj/item/I)
```
these procs will use oop soon i'm aware that this is awful

PinkyGatchabee commented:
kawaii

RemieRichards commented:
@dannno Well pete is a brilliant coder.

Giacom commented on code/modules/mob/living/simple_animal/friendly/keeper.dm:
```diff
@@ -0,0 +1,111 @@
+/mob/living/simple_animal/keeper
+	name = "keeper"
+	desc = "A keeper droid. They're kawaii as fuck and they'll repair up your shit."
```
muh immersion, could you make a sensible description?

Petethegoat replied:
NO

Iamgoofball replied:
plz dont ever use the word kawaii again

kthx

also yeah as great that description is i'm with giacom on this one. switch the description.

RemieRichards replied:
@Iamgoofball Atashi wa cho kawaii, anata wa baka desu!

Petethegoat committed:
Keeper fixes and improvements. New sprites, new description, laws, inhand sprites.

Petethegoat commented:
Updoot. Ready for testing now. There's probably a lot of shit that won't work, so the more we find the better.
![http://i.imgur.com/WRAxlrE.png](http://i.imgur.com/WRAxlrE.png)

Atskadan commented:
wow look at that tiny bucket
that tiny extinguisher
that tiny little dude
hey dude whats up how are you 
gonna go fight some fires hahahah ok little guy have fun

Atskadan commented on code/modules/mob/mob_movement.dm:
```diff
@@ -31,15 +31,10 @@
-		usr << "\red This mob type cannot drop items."
-	return
+	if(!usr.get_active_hand())
+		usr << "\red You have nothing to drop in your hand."
```
SPAN CLASSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Petethegoat replied:
aa AAAAAAAA

Petethegoat committed:
Updated a bunch of places so they use IsAdvancedToolUser().
Updated a bunch of "You don't got the dexterity!" messages with consistent span classes.

Added PASSTABLE flag for keepers, plus obj/turf vision.
Adds universal_speak, though they still can't talk outside of binary chat.
Removes atmospheric damage from keepers.
Adds keeper cannibalism.

Did something with gun sprites?? No idea what but it's probably awesome.

Iamgoofball commented:
can't automerge

did we touch the map at all in other PRs? as the food one barfed too

Atskadan commented:
it doesnt matter if you cant automerge because it says [DONT MERGE] in the title
its just because he hasnt pulled the most recent changes to his repo

Petethegoat committed:
Added keeper shells and colour picking.
Fixed clients being deleted by cannibalism, and fixed keepers holding onto items after they die.

Petethegoat committed:
Adds jobban checking. pAI bans now apply to keepers as well.

Neerti commented:
I hate the sprite personally but that's no reason to deny something.
:+1:

JJRcop commented:
pls hury and feenish

Petethegoat commented:
yur hur bluh

soon maybe

Atskadan commented:
> pls hury and feenish

ghost commented:
Please don't give them all access, they can vent crawl to get around plenty fine. There's no reason they'll need to use doors outside of maint doors in emergencies. All access will lead to grief and vent crawling is already all access.

Iamgoofball commented:
Can't ventcrawl if a pipe grts fucked up I think.

Plus if you're building a room and put a door in but no vents you're never able to get back in bcause no talking to other entities besides other mommis

The all access ensures 100% automacy

Iamgoofball commented:
Plus you'd alsi need to remove ventcrawling if you're going by the grief excuse

vista- commented:
@Petethegoat, any updates on this?

RemieRichards commented:
Bay are beating us https://github.com/Baystation12/Baystation12/pull/4941

RemieRichards commented:
If There is or if Pete can chalk me up some kind of "This isn't done but is wanted" Eg, a checklist or something.

But Yeah I wouldn't mind taking it on.

RemieRichards commented:
Pete's current code is Nice. This should be fun and easy to finish.

GunHog commented:
Is this still alive, or what?

Atskadan commented:
pete is 2whipped2code

RemieRichards commented:
Pete isn't ass, pete just needs a rest and a long holiday I think.
and Yeah, I probably will pick it up.

ghost commented:
looking forward to it!

ghost commented:
I am no longer looking forward to it.

Pete is now startoad-tier I guess.

Atskadan commented:
lol
no, he just quit

ghost commented:
But he also said he'd finish this. I guess we can wait till RR does it, but RR's in the middle of some other shit right now. I can w8 if it means some quality code behind this.

GunHog commented:
We reeeeeeally need someone else to take this up. :(

ghost commented:
Meh. RR finished his mutations thing, and is doing xenoarch right now. Again, I can wait.

RemieRichards commented:
> his

:/

ghost commented:
:\

GunHog commented:
Someone finish plz. That or loot Bay's version!

vista- commented:
_cough_ /vg/ _cough_ better _cough_ fun _cough_ theymadetheoriginalanyway
_cough_

It's the flu.

--Zsolt Hegyi
On 27 Jun 2014 21:50, "GunHog" notifications@github.com wrote:

> Someone finish plz. That or loot Bay's version!
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/NTStation/NTstation13/pull/397#issuecomment-47392820.

GunHog commented:
Or /vg/'s version! Just please get something moving! I feel as if this PR is never going to get finished, then closed because it was left to rot. It should not go the way of Dismemberment!

vista- commented:
The current code is not compatible with /VG/'s as they used a silicon-based
mob instead of a simple_animal. A full, new port would be required. If said
port begins I suggest that you close the PR.

--Zsolt Hegyi
On 27 Jun 2014 23:44, "GunHog" notifications@github.com wrote:

> Or /vg/'s version! Just please get something moving! I feel as if this PR
> is never going to get finished, then closed because it was left to rot. It
> should not go the way of Dismemberment!
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/NTStation/NTstation13/pull/397#issuecomment-47403750.

ghost commented:
Gun hog do it yourself seriously if you have such a huge boner for mommis

RemieRichards commented:
If anyone DOES attempt to port /vg/'s MoMMIs, I tried it the other day, and parts of the code are redundant (or in the case of 2 definitions, flat out wrong) and there's lots of copy pasted code.
and quite a few of the pieces of code present in /vg/'s MoMMIs is incompatible so, Enjoy. If anything I'd say the best bet is to finish Pete's code instead, it's more compatible.

ike709 commented:
So is this abandoned now or what?

RemieRichards commented:
I wonder why I'm closing this PR? hehe.

Anton-Kr commented:
RIP NTStation.
